lib/AIECreatePathfindFlows.cpp

```diff
@@ -217,6 +216,10 @@ class DynamicTileAnalysis {
       coordToShimMux[std::make_pair(col, row)] = switchboxOp;
       maxcol = std::max(maxcol, col);
       maxrow = std::max(maxrow, row);
+      // check if the shim tile has noc connection
+      if (!getTile(builder, col, row).isShimNOCTile())
+        switchboxOp.emitError("Attempting to route to or from ShimTile (")
+            << col << ", " << row << "), which has no NOC connection";
```

Comment:
If it's a precondition check, then why move it after the associated code?

Reply:
Agreed. This check has now been moved to immediately after when switchboxOp (a ShimMuxOp) has just been created. This error message is linked to that op in MLIR diagnostic, so as to take advantage of the "note" which points to this shim mux.